presto-main/src/main/java/com/facebook/presto/cost/PlanNodeStatsEstimateMath.java

```diff
@@ -113,4 +120,37 @@ private static SymbolStatsEstimate addColumnStats(SymbolStatsEstimate leftStats,
                 .setNullsFraction((nullsCountLeft + nullsCountRight) / newRowCount)
                 .build();
     }
+
+    public static PlanNodeStatsEstimate intersect(PlanNodeStatsEstimate left, PlanNodeStatsEstimate right)
+    {
+        checkArgument(new HashSet<>(left.getSymbolsWithKnownStatistics()).equals(new HashSet<>(right.getSymbolsWithKnownStatistics())));
+
+        PlanNodeStatsEstimate.Builder statsBuilder = PlanNodeStatsEstimate.builder();
+
+        double newRowCount = left.getOutputRowCount();
+        for (Symbol symbol : left.getSymbolsWithKnownStatistics()) {
+            SymbolStatsEstimate leftSymbolStats = left.getSymbolStatistics(symbol);
+            SymbolStatsEstimate rightSymbolStats = right.getSymbolStatistics(symbol);
+            StatisticRange leftRange = StatisticRange.from(leftSymbolStats);
+            StatisticRange rightRange = StatisticRange.from(rightSymbolStats);
+            StatisticRange intersection = leftRange.intersect(rightRange);
+
+            double overlapFraction = leftRange.overlapPercentWith(intersection);
```

Comment:
This seems to be broken. We can't treat left and right differently in context when the ordering doesn't matter. In other places like that we do either min, max or average.